geode-redis/src/main/java/org/apache/geode/redis/internal/executor/ExpireAtExecutor.java

```diff
@@ -62,18 +68,29 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
     long currentTimeMillis = System.currentTimeMillis();
 
     if (timestamp <= currentTimeMillis) {
-      command.setResponse(Coder.getIntegerResponse(context.getByteBufAllocator(), NOT_SET));
+      int result = NOT_SET;
+      RedisDataType redisDataType = context.getKeyRegistrar().getType(wKey);
+
+      if (redisDataType != null) {
+        regionProvider.getRegionForType(redisDataType).remove(wKey);
+        result = SET;
+      }
+
+      command.setResponse(
+          Coder.getIntegerResponse(
+              context.getByteBufAllocator(),
+              result));
       return;
     }
 
     long delayMillis = timestamp - currentTimeMillis;
 
     boolean expirationSet = false;
```

Comment:
You could remove the initialization if you want since it is redundant

Reply:
Finito.